main.go

```diff
@@ -400,17 +400,15 @@ func main() {
 		os.Exit(1)
 	}
 
-	if featuregate.CollectorUsesTargetAllocatorCR.IsEnabled() {
-		if err = controllers.NewTargetAllocatorReconciler(
-			mgr.GetClient(),
-			mgr.GetScheme(),
-			mgr.GetEventRecorderFor("targetallocator"),
-			cfg,
-			ctrl.Log.WithName("controllers").WithName("TargetAllocator"),
-		).SetupWithManager(mgr); err != nil {
-			setupLog.Error(err, "unable to create controller", "controller", "TargetAllocator")
-			os.Exit(1)
-		}
+	if err = controllers.NewTargetAllocatorReconciler(
```

Comment:
we should wrap this in an availability check to be sure the operator can function without the TA CRD.